public/index.html

```diff
@@ -10,6 +10,8 @@
     <title>Linkbrary</title>
   </head>
   <body>
-    <div id="root"></div>
+    <div id="root">
+      <div id="modal"></div>
+    </div>
```

Comment:
💊 
`root` div와 `modal` div가 같은 계층에 위치하고 있습니다. 
createPortal을 사용하여 모달을 modal div에 렌더링하려는 의도로 보여요
하지만 모달을 `root` div의 외부에 배치하는 것이 일반적으로 더 바람직합니다. 

이유: 모달의 z-index와 CSS 스타일링이 애플리케이션의 다른 요소들과 충돌할 수 있으며, 접근성(a11y)과 SEO에도 영향을 줄 수 있기 때문에

--- 
현재 구조에서 modal div가 root div 내부에 있어요! 
React DOM 트리 내에서 모달이 메인 애플리케이션의 하위 컴포넌트로 처리됨을 의미합니다. 
모달과 같은 오버레이 컴포넌트는 애플리케이션의 메인 콘텐츠와 별개로 관리되는 것이 좋습니다.


```suggestion
  <div id="root"></div>
  <div id="modal"></div>
```